superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -303,6 +321,32 @@ export default function DataSourcePanel({
           placeholder={t('Search Metrics & Columns')}
         />
         <div className="field-selections">
+          {datasource.type === DatasourceType.Dataset ||
```

Comment:
I think it would look cleaner if you made this check into its own function and then invoked it here.